src/safeds_runner/server/pipeline_manager.py

```diff
@@ -278,6 +300,81 @@ def runner_save_placeholder(placeholder_name: str, value: Any) -> None:
         current_pipeline.save_placeholder(placeholder_name, value)
 
 
+def runner_memoized_function_call(
+    function_name: str,
+    function_callable: typing.Callable,
+    parameters: list[Any],
+    hidden_parameters: list[Any],
+) -> Any:
+    """
+    Call a function that can be memoized and save the result.
+
+    If a function has been previously memoized, the previous result may be reused.
+
+    Parameters
+    ----------
+    function_name : str
+        Fully qualified function name
+    function_callable : typing.Callable
+        Function that is called and memoized if the result was not found in the memoization map
+    parameters : list[Any]
+        List of parameters for the function
+    hidden_parameters : list[Any]
+        List of hidden parameters for the function. This is used for memoizing some impure functions.
+
+    Returns
+    -------
+    Any
+        The result of the specified function, if any exists
+    """
+    if current_pipeline is not None:
```

Comment:
Turn this around (guard clause):

```py
if current_pipeline is None:
    return None # pragme: no cover

# ...
```

Removes the nesting and is more consistent with the DSL code.